src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -747,8 +747,22 @@ public BoundExpression Sequence(params BoundExpression[] parts)
         public BoundExpression Sequence(ImmutableArray<LocalSymbol> locals, params BoundExpression[] parts)
         {
             var builder = ArrayBuilder<BoundExpression>.GetInstance();
-            for (int i = 0; i < parts.Length - 1; i++) builder.Add(parts[i]);
+            for (int i = 0; i < parts.Length - 1; i++)
+            {
+                var part = parts[i];
+                if (LocalRewriter.ReadIsSideeffecting(part))
```

Comment:
Will change the name. I think in other places we use MakeXxxx convention, where arguments are lowered pieces of Xxxx, but the result is not necessarily Xxxx, but often a simpler semantically equivalent node.